src/main/java/org/jenkinsci/maven/plugins/hpi/MavenArtifact.java

```diff
@@ -57,7 +57,12 @@ public MavenArtifact(
     }
 
     public MavenProject resolvePom() throws ProjectBuildingException {
-        return builder.buildFromRepository(artifact,remoteRepositories,localRepository);
+        ProjectBuildingRequest buildingRequest =
+                new DefaultProjectBuildingRequest(session.getProjectBuildingRequest());
+        buildingRequest.setRemoteRepositories(remoteRepositories);
+        buildingRequest.setLocalRepository(localRepository);
+        buildingRequest.setProcessPlugins(false); // improve performance
```

Comment:
Disabling the processing of plugins has a significant effect on performance. I noticed build times had doubled compared to before this PR. Investigating further, I suspected the the Maven 2 APIs we were using before this PR never processed plugins. When I disabled plugin processing in fc9ae1eda1f49fbe68f73b8f48d88e42fd2f52c5, performance went back to its old levels. You can see the difference between build 1 and build 2 of this PR. Build 1 took 29 minutes; build 2 took 12 minutes. Builds of the main branch typically take 15 minutes. So this PR keeps performance at similar levels. I repeated the manual testing I had done in core and plugins, and processing plugins didn't seem to have any noticeable effect.